### PR TITLE
Adding persistence of logging preferences

### PR DESCRIPTION
**Pull request by bhariharan**
This stores the log levels and file size per component in `SharedPreferences`, so that these preferences are used even if the app is swapped out of memory or the device is restarted.

### CODE REVIEW

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java [hunk 1]
```diff
@@ -269,25 +272,53 @@ public void log(Level level, String tag, String message, Throwable e) {
...
-                        logLine = String.format(LOG_LINE_FORMAT, level, tag, message);
+                        logLine = String.format(LOG_LINE_FORMAT, curTime, level, tag, message);
                     }
                     fileLogger.addLogLine(logLine);
                 }
             }
         });
     }
+
+    private String getTimeFromUTC() {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java]: Also added logging of timestamp.

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/FileLogger.java [hunk 2]
```diff
@@ -102,11 +108,11 @@ public int getMaxSize() {
      *
      * @param size Maximum number of log lines.
      */
-    public synchronized void setMaxSize(int size) {
+    public void setMaxSize(int size) {
         if (size < 0) {
             size = 0;
         }
-        maxSize = size;
+        storeFileLoggerPrefs(size);
```
Comment (wmathurin) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/FileLogger.java]: Shouldn't maxSize also be set?
If I construct a FileLogger then call setMaxSize, woudn't the chosen maxSize only be used the next time I construct a FileLogger?

Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/FileLogger.java]: Nope - it should be used this time. It's set at the end of `storeFileLoggerPrefs` [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1421/files#diff-a6cbed71575822a567cf352b98d0d547R269).

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java [hunk 3]
```diff
@@ -49,8 +52,10 @@
 public class SalesforceLogger {
 
     private static final String TAG = "SalesforceLogger";
-    private static final String LOG_LINE_FORMAT = "LEVEL: %s, TAG: %s, MESSAGE: %s";
-    private static final String LOG_LINE_FORMAT_WITH_EXCEPTION = "LEVEL: %s, TAG: %s, MESSAGE: %s, EXCEPTION: %s";
+    private static final String LOG_LINE_FORMAT = "TIME: %s, LEVEL: %s, TAG: %s, MESSAGE: %s";
+    private static final String LOG_LINE_FORMAT_WITH_EXCEPTION = "TIME: %s, LEVEL: %s, TAG: %s, MESSAGE: %s, EXCEPTION: %s";
+    private static final String US_DATE_FORMAT = "yyyy.MM.dd G 'at' HH:mm:ss z";
```
Comment (wmathurin) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java]: Could we add the milliseconds (HH:mm:ss.SSS)?
Also should we just mirror the logcat date/time format e.g.
05-30 15:18:15.839 ?

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/FileLogger.java (hunk 2 above)
Comment (wmathurin) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/FileLogger.java]: 👍

**Comment by bhariharan** [comment]
Switched date format [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1421/commits/acc532690f07c6d94b508c65b8a2398ede136cca) and switched to 1 thread [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1421/commits/0a93a5bdf4a9831ebbcef44ad74de0c34df330d7).

**Pull request merged**